<a href="https://colab.research.google.com/github/Mathieu-Germain/LogicMojo-AI-ML-April26-Mathieu/blob/main/Practice_Maths_for_ML_practice_Notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

###Two dice are rolled. Find the probability that a) the sum is 7 b) both dice show the same number c) the product is even

### Probability Calculations for Two Dice Rolls

In [ ]:
from fractions import Fraction

# Define the sample space for rolling two dice
sample_space = [(d1, d2) for d1 in range(1, 7) for d2 in range(1, 7)]
total_outcomes = len(sample_space)

print(f"Total possible outcomes: {total_outcomes}")
# print(f"Sample space: {sample_space}") # Uncomment to see the full sample space

Total possible outcomes: 36


#### a) The sum is 7

In [ ]:
event_a_outcomes = [outcome for outcome in sample_space if sum(outcome) == 7]
probability_a = Fraction(len(event_a_outcomes), total_outcomes)
# print(f"Probability of a single event is: {probability_a}")
print(f"Outcomes where the sum is 7: {event_a_outcomes}")
print(f"Number of outcomes for event a: {len(event_a_outcomes)}")
print(f"Probability that the sum is 7: {probability_a} (approx. {probability_a:.4f})")

Probability of a single event is: 1/6
Outcomes where the sum is 7: [(1, 6), (2, 5), (3, 4), (4, 3), (5, 2), (6, 1)]
Number of outcomes for event a: 6
Probability that the sum is 7: 1/6 (approx. 0.1667)


#### b) Both dice show the same number

In [ ]:
event_b_outcomes = [outcome for outcome in sample_space if outcome[0] == outcome[1]]
probability_b = Fraction(len(event_b_outcomes), total_outcomes)

print(f"Outcomes where both dice show the same number: {event_b_outcomes}")
print(f"Number of outcomes for event b: {len(event_b_outcomes)}")
print(f"Probability that both dice show the same number: {probability_b} (approx. {probability_b:.4f})")

Outcomes where both dice show the same number: [(1, 1), (2, 2), (3, 3), (4, 4), (5, 5), (6, 6)]
Number of outcomes for event b: 6
Probability that both dice show the same number: 1/6 (approx. 0.1667)


#### c) The product is even

In [ ]:
# The product of two numbers is even if at least one of the numbers is even.
# This is equivalent to saying it's NOT (both numbers are odd).
# Outcomes where both are odd: (1,1), (1,3), (1,5), (3,1), (3,3), (3,5), (5,1), (5,3), (5,5)

event_c_outcomes = [outcome for outcome in sample_space if (outcome[0] * outcome[1]) % 2 == 0]
probability_c = Fraction(len(event_c_outcomes), total_outcomes)

print(f"Number of outcomes for event c (product is even): {len(event_c_outcomes)}")
print(f"Probability that the product is even: {probability_c} (approx. {probability_c:.4f})")

Number of outcomes for event c (product is even): 27
Probability that the product is even: 3/4 (approx. 0.7500)


Two dice are rolled. Find the probability that: a) the sum is 7 b) both dice show the same number c) the product is even
Sure, I can help you find those probabilities! I'll generate some Python code to simulate rolling two dice and calculate the probabilities for each scenario.

The probabilities have been calculated successfully!

## Here are the results:

a). The probability that the sum of the two dice is 7: 1/6 (approximately 0.1667) The outcomes for this event are: [(1, 6), (2, 5), (3, 4), (4, 3), (5, 2), (6, 1)]

b) The probability that both dice show the same number: 1/6 (approximately 0.1667) The outcomes for this event are: [(1, 1), (2, 2), (3, 3), (4, 4), (5, 5), (6, 6)]

c) The probability that the product of the two dice is even: 3/4 (approximately 0.7500) There are 27 outcomes where the product is even.

In [ ]:
A box contains 5 red and 3 blue balls. One ball is drawn at random. a) What is the probability it is red? b) If two balls are drawn (without replacement), what is the probability both are blue?

### Probability Calculations for Drawing Balls from a Box

In [ ]:
from fractions import Fraction

# Define the number of balls
num_red_balls = 5
num_blue_balls = 3
total_balls = num_red_balls + num_blue_balls

print(f"Total number of balls: {total_balls}")
print(f"Number of red balls: {num_red_balls}")
print(f"Number of blue balls: {num_blue_balls}")

Total number of balls: 8
Number of red balls: 5
Number of blue balls: 3


#### a) What is the probability it is red if one ball is drawn at random?

In [ ]:
probability_red = Fraction(num_red_balls, total_balls)

print(f"Probability of drawing one red ball: {probability_red} (approx. {probability_red:.4f})")

Probability of drawing one red ball: 5/8 (approx. 0.6250)


#### b) If two balls are drawn (without replacement), what is the probability both are blue?

In [ ]:
# Probability of drawing the first blue ball
prob_first_blue = Fraction(num_blue_balls, total_balls)

# After drawing one blue ball, there is one less blue ball and one less total ball
remaining_blue_balls = num_blue_balls - 1
remaining_total_balls = total_balls - 1

# Probability of drawing the second blue ball (given the first was blue)
prob_second_blue_given_first_blue = Fraction(remaining_blue_balls, remaining_total_balls)

# Probability of both being blue is the product of these two probabilities
probability_both_blue = prob_first_blue * prob_second_blue_given_first_blue

print(f"Probability of drawing the first blue ball: {prob_first_blue}")
print(f"Probability of drawing the second blue ball given the first was blue: {prob_second_blue_given_first_blue}")
print(f"Probability that both drawn balls are blue (without replacement): {probability_both_blue} (approx. {probability_both_blue:.4f})")

Probability of drawing the first blue ball: 3/8
Probability of drawing the second blue ball given the first was blue: 2/7
Probability that both drawn balls are blue (without replacement): 3/28 (approx. 0.1071)


### Generalized Probability Calculation Function

In [ ]:
from fractions import Fraction

def calculate_ball_drawing_probabilities(num_red_balls, num_blue_balls):
    """
    Calculates probabilities for drawing balls from a box.

    Args:
        num_red_balls (int): The number of red balls in the box.
        num_blue_balls (int): The number of blue balls in the box.

    Returns:
        tuple: A tuple containing:
            - Fraction: Probability of drawing one red ball.
            - Fraction: Probability of drawing two blue balls without replacement.
    """
    total_balls = num_red_balls + num_blue_balls

    if total_balls == 0:
        print("Error: No balls in the box.")
        return Fraction(0), Fraction(0)

    # a) Probability of drawing one red ball
    probability_red = Fraction(num_red_balls, total_balls)

    # b) Probability of drawing two blue balls (without replacement)
    if num_blue_balls < 2:
        probability_both_blue = Fraction(0)
    else:
        prob_first_blue = Fraction(num_blue_balls, total_balls)
        remaining_blue_balls = num_blue_balls - 1
        remaining_total_balls = total_balls - 1

        if remaining_total_balls == 0:
            probability_both_blue = Fraction(0)
        else:
            prob_second_blue_given_first_blue = Fraction(remaining_blue_balls, remaining_total_balls)
            probability_both_blue = prob_first_blue * prob_second_blue_given_first_blue

    return probability_red, probability_both_blue

In [ ]:
# Example usage: 7 red balls, 4 blue balls
example_num_red = 7
example_num_blue = 4

red_prob, both_blue_prob = calculate_ball_drawing_probabilities(example_num_red, example_num_blue)

print(f"For {example_num_red} red balls and {example_num_blue} blue balls:")
print(f"  a) Probability of drawing one red ball: {red_prob} (approx. {red_prob:.4f})")
print(f"  b) Probability of drawing two blue balls (without replacement): {both_blue_prob} (approx. {both_blue_prob:.4f})")

# Another example: 5 red balls, 3 blue balls (original problem)
print("\n--- Original problem verification ---")
original_red_prob, original_both_blue_prob = calculate_ball_drawing_probabilities(5, 3)
print(f"For 5 red balls and 3 blue balls:")
print(f"  a) Probability of drawing one red ball: {original_red_prob} (approx. {original_red_prob:.4f})")
print(f"  b) Probability of drawing two blue balls (without replacement): {original_both_blue_prob} (approx. {original_both_blue_prob:.4f})")

For 7 red balls and 4 blue balls:
  a) Probability of drawing one red ball: 7/11 (approx. 0.6364)
  b) Probability of drawing two blue balls (without replacement): 6/55 (approx. 0.1091)

--- Original problem verification ---
For 5 red balls and 3 blue balls:
  a) Probability of drawing one red ball: 5/8 (approx. 0.6250)
  b) Probability of drawing two blue balls (without replacement): 3/28 (approx. 0.1071)


#### Demonstrating the generalized function with an example

Okay, I see you have a new probability problem. I will generate the Python code to solve this for you, calculating the probabilities for drawing red and blue balls from the box.

The probabilities for drawing balls from the box have been calculated successfully!

Here are the results:

Total Balls: 8 (5 red, 3 blue)
a) The probability of drawing one red ball at random: 5/8 (approximately 0.6250)

b) The probability of drawing two blue balls without replacement: 3/28 (approximately 0.1071) (This is calculated as the probability of drawing the first blue ball (3/8) multiplied by the probability of drawing a second blue ball given the first was blue (2/7)).

A card is drawn from a deck. Find the probability that it is a) a heart, given that it is red b) a king, given that it is a face card

### Probability Calculations for Drawing Cards from a Deck

In [ ]:
from fractions import Fraction

# A standard deck has 52 cards.
# 4 suits: Hearts, Diamonds, Clubs, Spades
# 2 colors: Red (Hearts, Diamonds), Black (Clubs, Spades)
# Each suit has 13 cards: A, 2, 3, 4, 5, 6, 7, 8, 9, 10, J, Q, K

total_cards = 52

# Define relevant counts for the problem
num_red_cards = 26  # Hearts (13) + Diamonds (13)
num_hearts = 13
num_kings = 4       # One King per suit

# Face cards: Jack, Queen, King (3 per suit * 4 suits)
num_face_cards = 12 # J, Q, K of Hearts, Diamonds, Clubs, Spades

print(f"Total cards in a deck: {total_cards}")
print(f"Number of red cards: {num_red_cards}")
print(f"Number of hearts: {num_hearts}")
print(f"Number of face cards: {num_face_cards}")
print(f"Number of kings: {num_kings}")

Total cards in a deck: 52
Number of red cards: 26
Number of hearts: 13
Number of face cards: 12
Number of kings: 4


#### a) Probability that it is a heart, given that it is red P(Heart | Red)

In [ ]:
# The event is drawing a heart. The condition is that the card is red.
# The number of outcomes satisfying the condition (red cards) is 26.
# The number of outcomes satisfying both the event and the condition (red hearts) is 13.

# P(Heart | Red) = P(Heart and Red) / P(Red)
# P(Heart and Red) = Number of red hearts / Total cards = 13 / 52
# P(Red) = Number of red cards / Total cards = 26 / 52

# Alternatively, if we know it's red, our sample space reduces to the 26 red cards.
# Out of these 26 red cards, 13 are hearts.

probability_heart_given_red = Fraction(num_hearts, num_red_cards)

print(f"Number of red hearts: {num_hearts}")
print(f"Number of red cards: {num_red_cards}")
print(f"Probability it is a heart, given that it is red: {probability_heart_given_red} (approx. {probability_heart_given_red:.4f})")

Number of red hearts: 13
Number of red cards: 26
Probability it is a heart, given that it is red: 1/2 (approx. 0.5000)


#### b) Probability that it is a king, given that it is a face card P(King | Face Card)

In [ ]:
# The event is drawing a king. The condition is that the card is a face card.
# The number of outcomes satisfying the condition (face cards) is 12.
# The number of outcomes satisfying both the event and the condition (kings that are also face cards) is 4 (all kings are face cards).

# P(King | Face Card) = P(King and Face Card) / P(Face Card)
# P(King and Face Card) = Number of kings / Total cards = 4 / 52
# P(Face Card) = Number of face cards / Total cards = 12 / 52

# Alternatively, if we know it's a face card, our sample space reduces to the 12 face cards.
# Out of these 12 face cards, 4 are kings.

probability_king_given_face_card = Fraction(num_kings, num_face_cards)

print(f"Number of kings (which are also face cards): {num_kings}")
print(f"Number of face cards: {num_face_cards}")
print(f"Probability it is a king, given that it is a face card: {probability_king_given_face_card} (approx. {probability_king_given_face_card:.4f})")

Number of kings (which are also face cards): 4
Number of face cards: 12
Probability it is a king, given that it is a face card: 1/3 (approx. 0.3333)


##Two cards are drawn without replacement. Find the probability that a) the second card is a heart, given the first card is a heart b) the second card is a king, given the first card is a king c) draw 2 cards witout replacement, find prob of both being hearts

### Probability Calculations for Drawing Two Cards Without Replacement

In [ ]:
from fractions import Fraction

# Initial state of a standard deck
initial_total_cards = 52
initial_num_hearts = 13
initial_num_kings = 4

print(f"Initial total cards: {initial_total_cards}")
print(f"Initial number of hearts: {initial_num_hearts}")
print(f"Initial number of kings: {initial_num_kings}")

Initial total cards: 52
Initial number of hearts: 13
Initial number of kings: 4


#### a) The second card is a heart, given the first card is a heart

In [ ]:
# Event: First card drawn is a heart.
# State after first draw:
remaining_total_cards_a = initial_total_cards - 1 # 51 cards left
remaining_num_hearts_a = initial_num_hearts - 1   # 12 hearts left

probability_second_heart_given_first_heart = Fraction(remaining_num_hearts_a, remaining_total_cards_a)

print(f"Remaining hearts after first heart drawn: {remaining_num_hearts_a}")
print(f"Remaining total cards after first heart drawn: {remaining_total_cards_a}")
print(f"Probability that the second card is a heart, given the first is a heart: {probability_second_heart_given_first_heart} (approx. {probability_second_heart_given_first_heart:.4f})")

Remaining hearts after first heart drawn: 12
Remaining total cards after first heart drawn: 51
Probability that the second card is a heart, given the first is a heart: 4/17 (approx. 0.2353)


#### b) The second card is a king, given the first card is a king

In [ ]:
# Event: First card drawn is a king.
# State after first draw:
remaining_total_cards_b = initial_total_cards - 1 # 51 cards left
remaining_num_kings_b = initial_num_kings - 1     # 3 kings left

probability_second_king_given_first_king = Fraction(remaining_num_kings_b, remaining_total_cards_b)

print(f"Remaining kings after first king drawn: {remaining_num_kings_b}")
print(f"Remaining total cards after first king drawn: {remaining_total_cards_b}")
print(f"Probability that the second card is a king, given the first is a king: {probability_second_king_given_first_king} (approx. {probability_second_king_given_first_king:.4f})")

Remaining kings after first king drawn: 3
Remaining total cards after first king drawn: 51
Probability that the second card is a king, given the first is a king: 1/17 (approx. 0.0588)


#### c) Probability of drawing two hearts without replacement

In [ ]:
# Probability of the first card being a heart
prob_first_heart = Fraction(initial_num_hearts, initial_total_cards)

# Probability of the second card being a heart given the first was a heart (from part a)
# We can reuse the result from part a or recalculate it here for clarity:
# prob_second_heart_given_first_heart = Fraction(initial_num_hearts - 1, initial_total_cards - 1)

probability_both_hearts = prob_first_heart * probability_second_heart_given_first_heart

print(f"Probability of first card being a heart: {prob_first_heart}")
print(f"Probability of second card being a heart given first was heart: {probability_second_heart_given_first_heart}")
print(f"Probability of drawing two hearts without replacement: {probability_both_hearts} (approx. {probability_both_hearts:.4f})")

Probability of first card being a heart: 1/4
Probability of second card being a heart given first was heart: 4/17
Probability of drawing two hearts without replacement: 1/17 (approx. 0.0588)


###A factory has two machines, M1 and M2.
###M1 produces 60% of items, M2 produces 40%.
###Defective rates: 3% for M1, 5% for M2. If an item is found defective, what is the probability it was produced by M1?

To solve this problem, we use **Bayes' Theorem**, which allows us to find the probability of an event based on prior knowledge of conditions related to the event.

Here is the step-by-step breakdown:

### Step 1: Define the Events

* Let $M_1$ be the event that an item is produced by Machine 1.
* Let $M_2$ be the event that an item is produced by Machine 2.
* Let $D$ be the event that an item is defective.

### Step 2: Identify the Given Probabilities

* **Production Shares (Prior Probabilities):**
* $P(M_1) = 60\% = 0.60$
* $P(M_2) = 40\% = 0.40$


* **Defective Rates (Conditional Probabilities/Likelihoods):**
* Probability an item is defective *given* it came from $M_1$:  $P(D \mid M_1) = 3\% = 0.03$
* Probability an item is defective *given* it came from $M_2$:  $P(D \mid M_2) = 5\% = 0.05$



### Step 3: Find the Total Probability of a Defective Item, $P(D)$

To find the overall probability that any randomly selected item is defective, we use the **Law of Total Probability**. This counts defective items from both machines:


$$P(D) = [P(D \mid M_1) \times P(M_1)] + [P(D \mid M_2) \times P(M_2)]$$

Substitute the numbers into the formula:


$$P(D) = (0.03 \times 0.60) + (0.05 \times 0.40)$$

$$P(D) = 0.018 + 0.020$$

$$P(D) = 0.038$$

*(This means that 3.8% of all items produced in the factory are defective).*

---

### Step 4: Apply Bayes' Theorem

We want to find the conditional probability that the item came from Machine 1, *given* that we already know it is defective ($P(M_1 \mid D)$).

The formula for Bayes' Theorem is:


$$P(M_1 \mid D) = \frac{P(D \mid M_1) \times P(M_1)}{P(D)}$$

Substitute our calculated values into the formula:


$$P(M_1 \mid D) = \frac{0.03 \times 0.60}{0.038}$$

$$P(M_1 \mid D) = \frac{0.018}{0.038}$$

Now, simplify the fraction:


$$P(M_1 \mid D) = \frac{18}{38} = \frac{9}{19}$$

Convert to a decimal and percentage:


$$P(M_1 \mid D) \approx 0.4737 \text{ or } \mathbf{47.37\%}$$

### Final Answer

If an item is found to be defective, the probability that it was produced by Machine 1 is **$\frac{9}{19}$** (or approximately **$47.37\%$**).

#Exercise:
A certain disease affects 1% of the population. A test has:

Sensitivity (true positive rate) = 95%

Specificity (true negative rate) = 90% If a person tests positive, what is the probability they actually have the disease?

To solve this problem, we can use **Bayes' Theorem** or a simpler **hypothetical population method**. This problem is a classic example of how a relatively accurate medical test can yield counterintuitive results when the disease itself is rare (a phenomenon known as the **Base Rate Fallacy**).

Here are two different ways to understand and solve the problem:

---

### Method 1: The Hypothetical Population Method (Easiest to Visualize)

Let's imagine a population of **10,000 people**.

**Step 1: Divide the population into sick and healthy groups**

* **1% have the disease:** $10,000 \times 0.01 = \mathbf{100\text{ people are sick}}$.
* **The remaining 99% are healthy:** $10,000 - 100 = \mathbf{9,900\text{ people are healthy}}$.

**Step 2: Apply the test results to both groups**

* **Of the 100 sick people:** The test has a 95% sensitivity (true positive rate).
* $100 \times 0.95 = \mathbf{95\text{ True Positives}}$ (sick people who test positive).


* **Of the 9,900 healthy people:** The test has a 90% specificity (true negative rate). This means 90% test negative, but the remaining **10% will mistakenly test positive** (false positive rate).
* $9,900 \times 0.10 = \mathbf{990\text{ False Positives}}$ (healthy people who test positive).



**Step 3: Analyze the total pool of positive tests**
If you look at everyone who received a positive test result, your total pool is:


$$\text{Total Positive Tests} = 95 \text{ (True Positives)} + 990 \text{ (False Positives)} = \mathbf{1,085\text{ people}}$$

**Step 4: Find the final probability**
Out of those 1,085 people who tested positive, how many actually have the disease? Only the 95 true positives.


$$\text{Probability} = \frac{95}{1,085} \approx 0.08755 \text{ or } \mathbf{8.76\%}$$

---

### Method 2: Using Bayes' Theorem (Formal Probability)

#### 1. Define the Events

* $D$: The person has the disease.
* $H$: The person is healthy (does not have the disease).
* $Pos$: The person tests positive.

#### 2. Identify Given Values

* **Base Rate / Prior Probability:** $P(D) = 1\% = 0.01$
* **Complementary Probability:** $P(H) = 99\% = 0.99$
* **Sensitivity (True Positive Rate):** $P(Pos \mid D) = 95\% = 0.95$
* **Specificity (True Negative Rate):** $P(Neg \mid H) = 90\% = 0.90 \rightarrow$ Thus, the **False Positive Rate** is $P(Pos \mid H) = 10\% = 0.10$

#### 3. Find Total Probability of Testing Positive, $P(Pos)$

Using the Law of Total Probability:


$$P(Pos) = [P(Pos \mid D) \times P(D)] + [P(Pos \mid H) \times P(H)]$$

$$P(Pos) = (0.95 \times 0.01) + (0.10 \times 0.99)$$

$$P(Pos) = 0.0095 + 0.0990 = \mathbf{0.1085}$$

#### 4. Apply Bayes' Theorem

We want to find $P(D \mid Pos)$, the probability of having the disease given a positive test:


$$P(D \mid Pos) = \frac{P(Pos \mid D) \times P(D)}{P(Pos)}$$

$$P(D \mid Pos) = \frac{0.95 \times 0.01}{0.1085}$$

$$P(D \mid Pos) = \frac{0.0095}{0.1085} \approx \mathbf{0.0876}$$

---

### Final Answer

If a person tests positive, the probability that they actually have the disease is approximately **$8.76\%$** (or a fraction of $\frac{19}{217}$).

### Why is this so low?

Even though the test is highly accurate, the disease is very rare (1%). Because the healthy population is so large (99%), the small 10% error margin of the test generates **990 false positives**, which completely overwhelms the **95 true positives**. Therefore, most people who test positive in this scenario are actually completely healthy!

##Exercise:
There are three boxes:

Box A: 2 gold, 1 silver

Box B: 1 gold, 2 silver

Box C: 3 silver
A box is chosen at random and a coin (or ball) is drawn. If it is gold, what is the probability it came from Box A?

To solve this problem, we can use **Bayes' Theorem**, which updates the probability of an event (choosing Box A) based on new evidence (knowing the drawn coin is gold).

Here is the step-by-step breakdown:

### Step 1: Define the Events

* Let $A$ be the event that Box A is chosen.
* Let $B$ be the event that Box B is chosen.
* Let $C$ be the event that Box C is chosen.
* Let $G$ be the event that a gold coin is drawn.

### Step 2: Find the Prior Probabilities

Since a box is chosen completely at random out of three options, each box has an equal chance of being selected:

* $P(A) = \frac{1}{3}$
* $P(B) = \frac{1}{3}$
* $P(C) = \frac{1}{3}$

### Step 3: Find the Conditional Probabilities (Likelihoods)

Next, we calculate the probability of drawing a gold coin given that a specific box was chosen:

* **Box A** has 2 gold and 1 silver (3 total coins): $P(G \mid A) = \frac{2}{3}$
* **Box B** has 1 gold and 2 silver (3 total coins): $P(G \mid B) = \frac{1}{3}$
* **Box C** has 0 gold and 3 silver (3 total coins): $P(G \mid C) = \frac{0}{3} = 0$

### Step 4: Find the Total Probability of Drawing a Gold Coin, $P(G)$

Using the **Law of Total Probability**, we combine the probability of selecting each box with the probability of getting gold from that box:


$$P(G) = [P(G \mid A) \times P(A)] + [P(G \mid B) \times P(B)] + [P(G \mid C) \times P(C)]$$

Substitute the values:


$$P(G) = \left(\frac{2}{3} \times \frac{1}{3}\right) + \left(\frac{1}{3} \times \frac{1}{3}\right) + \left(0 \times \frac{1}{3}\right)$$

$$P(G) = \frac{2}{9} + \frac{1}{9} + 0 = \frac{3}{9} = \frac{1}{3}$$

So, the total probability of drawing a gold coin across the entire setup is $\frac{1}{3}$.

---

### Step 5: Apply Bayes' Theorem

We want to find the conditional probability that the box was Box A, *given* that the coin drawn is gold ($P(A \mid G)$).

The formula for Bayes' Theorem is:


$$P(A \mid G) = \frac{P(G \mid A) \times P(A)}{P(G)}$$

Substitute the values we calculated:


$$P(A \mid G) = \frac{\frac{2}{3} \times \frac{1}{3}}{\frac{1}{3}}$$

To simplify, since $\frac{1}{3}$ is in both the numerator and denominator, they cancel out:


$$P(A \mid G) = \frac{2}{3}$$

### Final Answer

If the drawn coin is gold, the probability it came from Box A is **$\frac{2}{3}$** (or approximately **$66.67\%$**).

Question 1:- What is the primary difference between a Z-test and a t-test?(Easy)
A) Z-test is used when the sample size is small; t-test is used for large samples.
B) Z-test is used when the population variance is known; t-test is used when the population variance is unknown.
C) Z-test is used for categorical data; t-test is used for continuous data.
D) Z-test is for paired samples; t-test is for independent samples.

### The correct answer is B) Z-test is used when the population variance is known; t-test is used when the population variance is unknown.

Detailed Explanation:
* *Z-test: You use a Z-test when  - you know the true population standard deviation ($\sigma$)—and by extension, the population variance ($\sigma^2$). Because the exact variance is known, the test statistic follows a standard normal distribution (Z-distribution). It is also typically applied when the sample size is large ($n \ge 30$), which justifies the assumption of normality via the Central Limit Theorem.

* *t-test: You use a t-test when the population variance is unknown. Instead, you have to estimate it using the sample standard deviation ($s$). Because estimating the variance from a sample introduces extra uncertainty, the test statistic follows a t-distribution. The t-distribution has heavier, fatter tails to account for this added variability, which is especially critical when working with small sample sizes ($n < 30$).

Why the other options are incorrect:

* A is reversed: Z-tests are typically used for large samples ($n \ge 30$), whereas t-tests are historically designed for small samples (though a t-test can also be used effectively on larger sample sizes).
* C is incorrect: Both Z-tests and t-tests are parametric tests designed for continuous data where you are comparing means (e.g., heights, test scores, revenue). Categorical data would typically require tests like Chi-Square.
* D is incorrect: Both Z-tests and t-tests have variations that can handle both paired (dependent) samples and independent samples.

* Question 2:
- What is the purpose of ANOVA (Analysis of Variance)?(Easy)

A) To test the difference between the means of two groups.

B) To test the difference between the means of three or more groups.

C) To determine the relationship between two variables.

D) To predict a dependent variable from an independent variable.

##The correct answer is B) To test the difference between the means of three or more groups.

##Why this is the case:
* ANOVA (Analysis of Variance): While the name focuses on "variance," its primary purpose is to compare the means of three or more independent groups to see if at least one of them is significantly different from the others.

* It accomplishes this by analyzing the variance between the groups relative to the variance within the groups.

* If you tried to use multiple t-tests to compare three or more groups (e.g., Group A vs. B, B vs. C, and A vs. C), you would drastically increase the risk of making a Type I error (a false positive). ANOVA solves this by testing all the groups simultaneously.

##Quick breakdown of the other options:
* A refers to a t-test: A two-sample t-test (or Z-test) is specifically used when you only have two groups to compare.
* C refers to Correlation: Determining the strength and direction of a relationship between two variables is the specific job of correlation analysis (like Pearson's $r$).
* D refers to Regression: Predicting a dependent variable based on one or more independent variables is the definition of linear or multiple regression analysis.

Question 3:

Which of the following is a correct interpretation of a p-value in hypothesis testing?(Easy)
  * A) The probability that the null hypothesis is true.
  * B) The probability of observing the data given that the null hypothesis is true.
  * C) The probability of making a Type I error.
  * D) The significance level of the test.

## The correct answer is B) The probability of observing the data given that the null hypothesis is true.

# Why this is the case:
  * The definition of a p-value: The p-value measures the probability of obtaining test results at least as extreme as the results actually observed, under the assumption that the null hypothesis ($H_0$) is correct.
  
  * It answers the question: "If the baseline assumption (the null) is perfectly true, how unusual or surprising is my sample data?".A very small p-value means the data would be highly unlikely under the null hypothesis, leading you to reject it.
  
  # Quick Breakdown of the other options:A is a common misconception:
  
  * A p-value is not the probability that the null hypothesis is true. In classical (frequentist) statistics, a hypothesis is either true or false; it does not have a probability assigned to it.
  
  * C refers to the Alpha ($\alpha$) level set by the researcher: The probability of committing a Type I error (falsely rejecting a true null hypothesis) is exactly equal to your pre-determined significance level ($\alpha$), usually set to 0.05 or 5%.
  *  D is incorrect: The significance level ($\alpha$) is a static threshold chosen before running the test (e.g., $\alpha = 0.05$). The p-value is an output calculated from the collected data. You compare the p-value to the significance level to make a decision.

uestion 4:- In hypothesis testing, what does a Type II error represent?(Easy)

A) Rejecting the null hypothesis when it is true.

B) Accepting the null hypothesis when it is false.

C) Rejecting the alternative hypothesis when it is true.

D) Accepting the alternative hypothesis when it is false.


# The correct answer is B) Accepting the null hypothesis when it is false.


(Note: In formal statistical terms, we say "failing to reject" the null hypothesis rather than "accepting" it, but the core concept remains the same.)

##Why this is the case:
A Type II error occurs when there is a real effect or difference in the population (meaning the null hypothesis is actually false), but your test fails to detect it. Think of it as a false negative.

##Quick Breakdown of the other options:

* A represents a Type I error: Rejecting the null hypothesis when it is actually true is a false positive (discovering an "effect" that doesn't really exist).


* C and D are incorrect phrasing: Hypothesis testing is structurally set up to frame decisions strictly around the null hypothesis (either rejecting it or failing to reject it), rather than formally "accepting or rejecting the alternative".